### PR TITLE
Feature/1066/attribute descriptors for svn tokeii raw

### PR DESCRIPTION
**Pull request by phanlezz**
# Adds AttributeDescriptors to svn and tokei

RawTextParser does not need descriptors, because the imported metrics are arbitrary.
Refactored more tests to be junit5.
Updated some deprecated code. Removed some unused.

Issue: #1066

### CODE REVIEW

**Review by MW-Friedrich — CHANGES_REQUESTED** [review]

File: analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/converter/ProjectConverter.kt [hunk 1]
```diff
@@ -54,8 +55,9 @@ class ProjectConverter(private val containsAuthors: Boolean) {
         val metrics = metricsFactory.createMetrics()
         projectBuilder.addAttributeTypes(AttributeTypesFactory.createNodeAttributeTypes(metrics))
         projectBuilder.addAttributeTypes(AttributeTypesFactory.createEdgeAttributeTypes(metrics))
+        projectBuilder.addAttributeDescriptions(getAttributeDescriptors())
 
-        return projectBuilder.build()
+        return projectBuilder.build(true)
```
[the remark this answers is not in the document]
Comment (MW-Friedrich) [reply analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/converter/ProjectConverter.kt]: Aaaand already outdated. Thanks for the change! :D

**Review by MW-Friedrich — APPROVED** [review]
LGTM! 🚀

File: analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/AttributeDescriptors.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    descriptors["number_of_renames"] = AttributeDescriptor(title = "Number of Renames", description = "Number of file renames", link = ghLink)
+    descriptors["weeks_with_commits"] = AttributeDescriptor(title = "Weeks with Commits", description = "Number of weeks in which commits occurred", link = ghLink)
+    descriptors["number_of_authors"] = AttributeDescriptor(title = "Number of Authors", description = "Number of authors", link = ghLink)
+    descriptors["temporal_coupling"] = AttributeDescriptor(title = "Temporal Coupling", description = "The degree of temporal coupling between two files (>=35%)", link = ghLink)
+    descriptors["successive_weeks_with_commits"] = AttributeDescriptor(title = "Successive Weeks with Commits", description = "Number of successive weeks in which the file was included in a commit", link = ghLink)
+    descriptors["range_of_weeks_with_commits"] = AttributeDescriptor(title = "Week Range of Commits", description = "Average number of weeks between commits", link = ghLink)
+
+    return descriptors.toMap()
```
Comment (BridgeAR) [inline analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/AttributeDescriptors.kt:19]:
```suggestion
    val ghLink = "https://github.com/MaibornWolff/codecharta/blob/main/analysis/import/SVNLogParser/README.md"

    return mapOf(
      "age_in_weeks" to AttributeDescriptor(title = "Age in Weeks", description = "File age in weeks", link = ghLink),
      "number_of_commits" to AttributeDescriptor(title = "Number of Commits", description = "Number of commits", link = ghLink),
      // ...
    )
```

File: analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt [hunk 3]
```diff
@@ -35,15 +35,15 @@ class ProjectMetricsTest {
...
+        Assertions.assertThat(projectMetrics.projectMetrics["foo"]!!.fileMetrics).isEqualTo(fileMetricMap.fileMetrics)
     }
 
     @Test
     fun `getFile MetricMap gets the metricMap for given file`() {
         val fileMetricMap = FileMetricMap().add("mcc", 99)
         projectMetrics.addFileMetricMap("foo", fileMetricMap)
 
-        Assertions.assertThat(projectMetrics.getFileMetricMap("foo")).isEqualTo(fileMetricMap)
+        Assertions.assertThat(projectMetrics.getFileMetricMap("foo")!!.fileMetrics).isEqualTo(fileMetricMap.fileMetrics)
```
Comment (BridgeAR) [inline analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt:46]: The former seems to validate more, no?

File: analysis/import/TokeiImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/tokeiimporter/AttributeDescriptors.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package de.maibornwolff.codecharta.importer.tokeiimporter
+
+import de.maibornwolff.codecharta.model.AttributeDescriptor
+
+fun getAttributeDescriptors(): Map<String, AttributeDescriptor> {
+    val descriptors = mutableMapOf<String, AttributeDescriptor>()
```
Comment (BridgeAR) [inline analysis/import/TokeiImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/tokeiimporter/AttributeDescriptors.kt:6]: See above

**Review by BridgeAR — APPROVED** [review]
LGTM % comments

File: analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt (hunk 3 above)
Comment (phanlezz) [reply analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt]: Good point, I will reverse this change.

File: analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/AttributeDescriptors.kt (hunk 2 above)
Comment (phanlezz) [reply analysis/import/SVNLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/AttributeDescriptors.kt]: Included as suggested

File: analysis/import/SVNLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/input/VersionControlledFileTest.kt [hunk 5]
```diff
@@ -160,12 +159,12 @@ class VersionControlledFileTest {
         // then
         assertThat(versionControlledFile.filename).isEqualTo(oldFilename)
         assertThat(versionControlledFile.actualFilename).isEqualTo(filename)
-        assertThat(versionControlledFile.markedDeleted()).isFalse()
+        assertThat(versionControlledFile.markedDeleted()).isFalse
```
Comment (BridgeAR) [inline analysis/import/SVNLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/input/VersionControlledFileTest.kt:162]: Is this not a function call anymore?

File: analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt [hunk 6]
```diff
@@ -26,7 +26,7 @@ class ProjectMetricsTest {
 
         addMetricToFileInProject(projectMetrics, "foo", "mcc", 99)
 
-        Assertions.assertThat(projectMetrics.projectMetrics["foo"]).isEqualToComparingFieldByField(expected)
+        Assertions.assertThat(projectMetrics.projectMetrics["foo"]!!.fileMetrics).isEqualTo(expected.fileMetrics)
```
Comment (BridgeAR) [inline analysis/import/SourceCodeParser/src/test/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metrics/ProjectMetricsTest.kt:29]: Seems like we compare less things in these tests now? Could be change that again?

File: analysis/import/SVNLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/input/VersionControlledFileTest.kt (hunk 5 above)
Comment (phanlezz) [reply analysis/import/SVNLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/svnlogparser/input/VersionControlledFileTest.kt:162]: It is an attribute provided by the assertion, afaik

**Pull request merged**